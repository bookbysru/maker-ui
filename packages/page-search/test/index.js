import React from 'react'
import { render, fireEvent, wait, cleanup } from '@testing-library/react'
import renderer from 'react-test-renderer'

import PageSearch from '../src'

const renderJSON = e => renderer.create(e).toJSON()

jest.useFakeTimers()
afterEach(cleanup)

const demoText =
  'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis lorem nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.'

const setup = () => {
  const utils = render(
    <>
      <PageSearch />
      <p>{demoText}</p>
    </>
  )
  const input = utils.getByPlaceholderText('Search this page...')
  return { input, ...utils }
}

test('PageSearch renders', () => {
  const json = renderJSON(<PageSearch />)
  expect(json).toMatchSnapshot()
})

test('PageSearch adds marks to document on search', () => {
  const { input } = setup()

  expect(input.value).toBe('')
  fireEvent.change(input, { target: { value: 'Lorem' } })
  expect(input.value).toBe('Lorem')

  wait(
    () => {
      expect(document.querySelectorAll('.search-mark').length).toBe(2)
    },
    { timeout: 1000 }
  )
})
