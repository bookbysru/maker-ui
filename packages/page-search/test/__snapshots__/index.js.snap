// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageSearch renders 1`] = `
.emotion-1 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  position: relative;
  top: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-0 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
}

<div
  className="emotion-1"
>
  <form
    aria-label="On this page"
    className="emotion-0"
    role="search"
  >
    <input
      name="search"
      onChange={[Function]}
      placeholder="Search this page..."
      type="search"
      value=""
    />
  </form>
  <button
    aria-label="Previous result"
    onClick={null}
    title="Previous result"
  >
    Prev
  </button>
  <button
    aria-label="Next result"
    onClick={null}
    title="Next result"
  >
    Next
  </button>
  <button
    aria-label="Clear search"
    onClick={[Function]}
    title="Clear"
  >
    Clear
  </button>
</div>
`;
